components/brave_wallet/browser/bitcoin/bitcoin_rpc.cc

```diff
@@ -0,0 +1,241 @@
+}
+
+void BitcoinRpc::GetAddressHistory(const std::string& network_id,
+                                   const std::string& address,
+                                   const uint32_t max_block_height,
+                                   const std::string& last_seen_txid,
+                                   GetAddressHistoryCallback callback) {
+  GURL network_url = BaseRpcUrl(network_id);
+
+  auto internal_callback = base::BindOnce(
+      &BitcoinRpc::OnGetAddressHistory, weak_ptr_factory_.GetWeakPtr(),
+      max_block_height, std::move(callback));
+  RequestInternal(true,
+                  MakeAddressHistoryUrl(network_url, address, last_seen_txid),
+                  std::move(internal_callback));
+}
+
+void BitcoinRpc::OnGetAddressHistory(const uint32_t max_block_height,
+                                     GetAddressHistoryCallback callback,
+                                     APIRequestResult api_request_result) {
+  if (!api_request_result.Is2XXResponseCode()) {
+    std::move(callback).Run(base::unexpected("Unexpected HTTP result code"));
+    return;
+  }
+
+  auto* items = api_request_result.value_body().GetIfList();
+  if (!items) {
+    std::move(callback).Run(base::unexpected("Json response is not an arary"));
+    return;
+  }
+
+  std::vector<Transaction> result;
+
+  for (auto& item : *items) {
+    auto transaction = Transaction::FromRpcValue(item);
+    if (!transaction) {
+      std::move(callback).Run(base::unexpected("Invalid transaction dict"));
+      return;
+    }
+    if (transaction->block_height > max_block_height) {
+      continue;
+    }
+
+    result.push_back(std::move(*transaction));
+  }
+
+  std::move(callback).Run(base::ok(result));
+}
+
+void BitcoinRpc::PostTransaction(const std::string& network_id,
+                                 const std::vector<uint8_t>& transaction,
+                                 PostTransactionCallback callback) {
+  GURL network_url = BaseRpcUrl(network_id);
+  auto payload = base::HexEncode(transaction);
+
+  auto internal_callback =
+      base::BindOnce(&BitcoinRpc::OnPostTransaction,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  auto conversion_callback = base::BindOnce(&ConvertPlainStringToJsonArray);
+  api_request_helper_->Request("POST", MakePostTransactionUrl(network_url),
+                               payload, "", true, std::move(internal_callback),
+                               {}, -1u, std::move(conversion_callback));
+}
+
+void BitcoinRpc::OnPostTransaction(PostTransactionCallback callback,
+                                   APIRequestResult api_request_result) {
+  if (!api_request_result.Is2XXResponseCode()) {
+    std::move(callback).Run(base::unexpected("Unexpected HTTP result code"));
+    return;
+  }
+
+  auto* list = api_request_result.value_body().GetIfList();
+  if (!list) {
+    std::move(callback).Run(base::unexpected("Json response is not an arary"));
```

Comment:
Consider using error codes if this strings need localization

Reply:
Going to handle error logging and localization closer to feature almost ready stage. Not sure actually if these errors should(or could) be shown to user and worth localization efforts.